src/fields/date/date.ts

```diff
@@ -50,6 +52,46 @@ export const date: IFieldGenerator<IDateSchema> = (id, { dateFormat = "uuuu-MM-d
 				.test("not-past", notPastRule?.["errorMessage"] || ERROR_MESSAGES.DATE.CANNOT_BE_PAST, (value) => {
 					if (!isValidDate(value, dateFormatter) || !notPastRule?.["notPast"]) return true;
 					return !LocalDate.parse(value, dateFormatter).isBefore(LocalDate.now());
+				})
+				.test("min-date", undefined, (value, context) => {
+					let minDate: LocalDate;
+					try {
+						minDate = LocalDate.parse(minDateRule?.["minDate"], dateFormatter);
```

Comment:
could `minDate` be parsed outside of the test? cmiiw but it's not a dynamic value. (and if the minDate format is invalid we could throw straight away)

then the error message can still be passed in to `.test()` as the second arg without having to call `context.createError()`

Reply:
yeah originally I was intending to make it more readable by parsing the dates in the test method but this has overheads indeed. have moved the parsing outside the tests instead.